Add support for the BCM2711 (#15188)
apache/nuttx: Documentation/platforms/arm64/bcm2711/boards/raspberrypi-4b/index.rst

===============
Raspberry Pi 4B
===============

The `Raspberry Pi 4B <https://www.raspberrypi.com/products/raspberry-pi-4-model-b/specifications/>`_ is an ARM64
hobbyist board created by Raspberry Pi.

.. figure:: raspberrypi-4b.png
   :figwidth: 70%
   :align: center
   :alt: Raspberry Pi 4B board

Features
=========

- Broadcom BCM2711 @1.8GHz
- 1, 2, 4 and 8GB LPDDR4-3200 SDRAM models
- 2.4GHz and 5.0GHz IEEE 802.11ac wireless
- Bluetooth 5.0
- Gigabit Ethernet
- 2 USB 3.0 ports
- 2 USB 2.0 ports
- 2 micro-HDMI ports (4kp60)
- 2-lane MIPI DSI display port
- 2-lane MIPI CSI camera port
- 4-pole stereo audio and composite video port
- Micro SD card slot

ARM64 Toolchain
===============

Before building NuttX for the Raspberry Pi 4B, download the ARM64 Toolchain for
**AArch64 Bare-Metal Target** ``aarch64-none-elf`` from
`Arm GNU Toolchain Downloads <https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads>`_.

> (Skip the section for Beta Releases)

Reviewer: ```suggestion
(Skip the section for Beta Releases.)
```

Print the partition table on the card with `p` to see what's there. If anything appears, continue to use the `d` command
to remove all partitions.

- `o` to create a new, empty DOS partition table
- `n` to create a new partition
- `p` to make it primary
- Hit enter to select the default partition of "1"
- Hit enter for the default start and end sizes, which will use the full SD card size
- `t` to change the type of the partition (hit enter to select default of partition 1)
- `c` as the type, which is for Windows FAT32
- `a` to mark the partition as bootable
- `w` to write all the changes and save

Now when you run `lsblk`, you should see `/dev/sda1` (or an equivalent for your SD card). That is the new partition just
created. Running the following command will then format the SD card to an empty FAT32 file system.

.. code:: console

   $ sudo mkfs.vfat /dev/sda1

Once this completes, you can copy all of the aforementioned boot files, `nuttx.bin` and `config.txt` to your SD card in
your preferred way (through a file explorer or by using `mount`).

Once all the files are copied, you can then eject the SD card and insert it onto your Raspberry Pi. The default console

> is the Mini UART, which requires a `USB to TTL serial converter cable <https://www.adafruit.com/product/954>`_ to read.

Reviewer: ```suggestion
is the Mini UART, which requires an adapter such as `USB to TTL serial converter cable <https://www.adafruit.com/product/954>`_ to read.
```

You should connect the ground to one of the Pi's ground pins, and then connect the RX to GPIO 14 and TX to GPIO 15. **Do

> not connect the red power cable**.

Reviewer: ```suggestion
not connect the red power wire**.
```

Once the converter is connected and plugged into your host computer, you can open up a serial terminal of your choice. I

> use Minicom. Then, power your Raspberry Pi 4 with a USB-C cable and wait for the Pi to boot and the NSH prompt to appear

Reviewer: ```suggestion
use Minicom. Then, power your Raspberry Pi 4b with a USB-C cable and wait for the Pi to boot and the NSH prompt to appear
```
